NewHorizons/Builder/Props/DetailBuilder.cs

```diff
@@ -390,5 +393,86 @@ public void Start()
                 Destroy(this);
             }
         }
+
+        // TODO: simulate in sector
+        // BUG: detector collider is not included in groups
+        // TODO: mass
+        private class AddPhysics : MonoBehaviour
+        {
+            private IEnumerator Start()
+            {
+                yield return new WaitForSeconds(.1f);
+
+                var parentBody = GetComponentInParent<OWRigidbody>();
+
+                foreach (var meshCollider in GetComponentsInChildren<MeshCollider>(true))
+                    // hack. doesnt work for all meshes but seems to for most
+                    if (!meshCollider.isTrigger)
+                        meshCollider.convex = true;
+
+                var owRigidbody = gameObject.AddComponent<OWRigidbody>();
+                owRigidbody.SetVelocity(parentBody.GetPointVelocity(transform.position));
+
+                var detector = new GameObject("Detector");
+                detector.transform.SetParent(transform, false);
+                detector.layer = LayerMask.NameToLayer("AdvancedDetector");
+                detector.tag = "DynamicPropDetector";
+                detector.AddComponent<SphereCollider>();
+                detector.AddComponent<DynamicForceDetector>();
+                detector.AddComponent<DynamicFluidDetector>();
+
+                var impactSensor = gameObject.AddComponent<ImpactSensor>();
+                var impactAudio = new GameObject("ImpactAudio");
+                impactAudio.transform.SetParent(transform, false);
+                var audioSource = impactAudio.AddComponent<AudioSource>();
+                audioSource.maxDistance = 30;
+                audioSource.dopplerLevel = 0;
+                audioSource.rolloffMode = AudioRolloffMode.Custom;
+                audioSource.playOnAwake = false;
+                audioSource.spatialBlend = 1;
+                var owAudioSource = impactAudio.AddComponent<OWAudioSource>();
+                owAudioSource._audioSource = audioSource;
+                owAudioSource.SetTrack(OWAudioMixer.TrackName.Environment);
```

Comment:
object audio impact adds owaudiosource/audiosource and sets them up properly already